DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/DelegateCourses/DelegateCoursesViewModel.cs

```diff
@@ -0,0 +1,59 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates.DelegateCourses
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Models.Courses;
+    using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
+    using DigitalLearningSolutions.Web.ViewModels.TrackingSystem.CourseSetup;
+
+    public class DelegateCoursesViewModel : BaseSearchablePageViewModel
+    {
+        public DelegateCoursesViewModel(
+            IEnumerable<CourseStatisticsWithAdminFieldResponseCounts> courses,
+            IEnumerable<string> categories,
+            IEnumerable<string> topics,
+            string? searchString,
+            string sortBy,
+            string sortDirection,
+            string? filterBy,
+            int page,
+            int? itemsPerPage
+        ) : base(
+            searchString,
+            page,
+            true,
+            sortBy,
+            sortDirection,
+            filterBy,
+            itemsPerPage ?? DefaultItemsPerPage,
+            "Search courses"
+        )
+        {
+            var sortedItems = GenericSortingHelper.SortAllItems(
+                courses.AsQueryable(),
+                sortBy,
+                sortDirection
+            );
+            var filteredItems = FilteringHelper.FilterItems(sortedItems.AsQueryable(), filterBy).ToList();
+            var searchedItems = GenericSearchHelper.SearchItems(filteredItems, SearchString).ToList();
+            var paginatedItems = GetItemsOnCurrentPage(searchedItems);
+
+            MatchingSearchResults = searchedItems.Count;
+            SetTotalPages();
```

Comment:
Can you use methods on the base class(es) instead? That simplifies it a bit. `PaginateItems()` covers the last three lines here.
(For searching you may as well use the searchHelper - you can't simplify it much more)